test/functional/feature_staking_free_tx_preference.py

```diff
@@ -0,0 +1,160 @@
+            self.sync_all()
+
+
+    def free_tx_trasaction_preference(self):
+        starting_height = 200
+        self.deposit_value = 6
+        self.free_tx_base_value = 10
+        free_tx_value = 9
+        free_tx_amount = free_tx_value * COIN
+        amount_of_tested_free_tx = int(self.free_tx_testing_proportion * self.blockmaxweight)
+        node0_height = self.nodes[0].getblockcount()
+        node1_height = self.nodes[1].getblockcount()
+        assert node0_height == node1_height == starting_height, f'Starting nodes height different than ' \
+                                                                f'{starting_height, node0_height, node1_height}'
+
+        # generate blocks on node 0, then sync nodes and check height and staking balance
+        addr1 = self.nodes[0].getnewaddress()
+        addr2 = self.nodes[0].getnewaddress()
+        dummy_addresses = [self.nodes[1].getnewaddress() for _ in range(60)]
+
+        self.nodes[0].generatetoaddress(amount_of_tested_free_tx*2, addr2)
+        self.sync_all()
+
+        # assure that staking balance is the same on both nodes
+        node0_staking_balance = self.get_staking_pool_balance(node_num=0)
+        node1_staking_balance = self.get_staking_pool_balance(node_num=1)
+        assert node0_staking_balance == node1_staking_balance
+
+        node0_height = self.nodes[0].getblockcount()
+        node1_height = self.nodes[1].getblockcount()
+        assert node0_height == node1_height, 'Difference in nodes height'
+
+        ##############################################
+        #                                            #
+        #            Test only free tx               #
+        #                                            #
+        ##############################################
+
+
+        self.generate_inputs_for_free_txs(amount_of_tested_free_tx, addr2)
+
+        self.assert_mempool_leftovers(0)
+        # load free txs into mempool
+        free_tx_list = []
+        for _ in range(amount_of_tested_free_tx):
+            free_tx_id = self.send_free_tx(dummy_addresses[:5], free_tx_amount, 0, addr2)
+            assert free_tx_id is not None
+            free_tx_list.append(free_tx_id)
+
+        self.nodes[0].generate(1)
+        # Assert test case with only free transactions we need to be between 24% and 25%
+        tx_in_block = self.assert_block_tx_preference_only_free(free_tx_list)
+
+        tx_in_block_without_coinbase_tx = tx_in_block -1
+        self.assert_mempool_leftovers(amount_of_tested_free_tx - tx_in_block_without_coinbase_tx)
+
+        ##############################################
+        #                                            #
+        #           Test free and paid tx            #
+        #                                            #
+        ##############################################
+
+        self.generate_inputs_for_free_txs(amount_of_tested_free_tx, addr2)
+
+        self.assert_mempool_leftovers(0)
```

Comment:
I assume this is temporary - we want to have mempool leftovers to test if the preference worked correctly

Reply:
It is to test if we have empty mempool before sending transactions for the test